Docs: New doc pages for panels Stat, Gauge & Bar Gauge (#22335)
grafana/grafana: docs/sources/features/panels/bar_gauge.md

+++
draft = "false"
date = "2020-02-19"
title = "Bar gauge panel"
description = "Bar gauge panel docs"
keywords = ["grafana", "enter", "keywords", "here"]
type = "docs"
+++

# Bar gauge panel

{{< docs-imagebox img="/img/docs/v66/bar_gauge_cover.png" max-width="1025px" caption="Stat panel" >}}

This panel can show one or more bar gauges depending on how many series, rows or columns your query returns. The
thresholds control the bar & value color.

## Display options

* Show

>   * `Calculation` - Calculate a show a single value like min or max based on all rows

Reviewer: ```suggestion
  * `Calculation` - Renders a single value like min or max based on all rows
```

>   * `All values` - Render a seperate stat for every row

Reviewer: ```suggestion
  * `All values` - Render a seperate stat for every row
```

* Calc
  * Specify calculation / reducer function. Since this panel is designed to only show a single value Grafana needs to

>   * know how to reduce a fields many values to a single value.

Reviewer: ```suggestion
  know how to reduce a fields many values to a single value.
```

* Orientation

>   * `Horizontal` - The bar will streth horizontally from left to right

Reviewer: ```suggestion
  * `Horizontal` - The bar will stretch horizontally from left to right
```

>   * `Vertical` - The bar will streth vertically from top to bottom

Reviewer: ```suggestion
  * `Vertical` - The bar will stretch vertically from top to bottom
```

* Mode
  * `Gradient` - The threshold levels define a gradient.

>   * `Retro LCD` - The gauge is split up in small cells that are lit or unlit

Reviewer: ```suggestion
  * `Retro LCD` - The gauge is split up in small cells that are lit or unlit.
```

  * `Basic` - Single color based on the matching threshold.

Retro LCD example:

{{< docs-imagebox img="/img/docs/v66/bar_gauge_lcd.png" max-width="1025px" caption="Stat panel" >}}

### Field

* `Title` - When multiple stats are shown this field controls the title in each stat. By default this is the series name
 and field name. You can use expressions like `${__series.name}` or `${__field.name}` to use only series name or field
 name in title or `${__cell_2}` to refer to other fields (2 being field/column with index 2).
* `Min` - The minimum value, leave blank for auto calculation based on all series & fields. Used by Graph to set y-axis min.
* `Max` - The maximum value, leave blank for auto calculation based on all series & fields. Used by Graph to set y-axis max.
* `Decimals` - Number of decimals to render value with. Leave empty for Grafana to automatically figure out the best
 number of decimals to use.

### Thresholds

> Define thresholds that will define the color of either the value or the background depending on your `Color` display option. The

Reviewer: ```suggestion
Define thresholds that will set the color of either the value or the background depending on your `Color` display option. The
```

> thresholds are automatically sorted from lowerst value to highest. The `Base` value represents minus infinity.

Reviewer: ```suggestion
thresholds are automatically sorted from lowest value to highest. The `Base` value represents minus infinity.
```
